neurons/validator.py

```diff
@@ -176,33 +172,39 @@ async def forward(self):
                 deserialize=False,
                 timeout=300
             )
-            # Getting the responses
+            # Processing miner output into rewards
             for j in range(0, len(responses)):
                 if responses[j].miner_response is not None:
                     if task_string.endswith('speech'):
                         miner_output_data = audio_decode(responses[j].miner_response)
                     else:
                         miner_output_data = responses[j].miner_response
-                        
-                    bt.logging.info(f'DECODED OUTPUT DATA: {miner_output_data}')
                     
-                    successful.append([miner_output_data, j])
+                    results.append(
+                        int(self.process_validator_output(
+                            miner_output_data,
+                            sample_request['output'],
+                            task_string
+                        ) * 1000) # 'numpy.float64' object cannot be interpreted as integer
+                    )
                 else:
-                    bt.logging.warning(f"Miner {j} failed to respond.")
+                    results.append(
+                        0
+                    )
         except Exception as e:
             bt.logging.error(f"Failed to query miners with exception: {e}")
-        # Rewarding the miners
-        bt.logging.info(f"successful:{successful}")
-        results = []
-        for i in range(len(successful)):
-            results.append([successful[i][1], self.process_validator_output(successful[i][0], sample_request['output'], task_string)])
-            # Updating the scores
-            self.update_scores(results[i][1], results[i][0])
+        
+        # Updating the scores
+        bt.logging.debug(f"Results: {results}")
+        self.update_scores(np.array(results), healthy_axon_uids)    
+            
         # Set weights
         self.now = time.time()
         if self.now % 10 == 0:
             self.set_weights()
         
+    # ? Regardless of the type, this is the "sum" of one and divided by 1?
```

Comment:
What's the purpose of this comment?  To review later?

Reply:
Yeah, for discussion

Reply:
I think previously we/I/you were using #TODO: nomenclature or tagging if you will.  Is this # ? Format something special?

Reply:
I use an extension called better comments on cursor/vscode and it's super useful. This `# ?` format just gives it a different color, a light blue.